src/vcpkg/base/downloads.cpp

```diff
@@ -883,6 +884,110 @@ namespace vcpkg
         return true;
     }
 
+    bool store_to_azblob_cache(DiagnosticContext& context,
+                               const Filesystem& fs,
+                               StringView raw_url,
+                               const SanitizedUrl& sanitized_url,
+                               View<std::string> headers,
+                               const Path& file,
+                               size_t file_size,
+                               size_t max_single_write,
+                               size_t block_size)
+    {
+        if (file_size <= max_single_write)
+            return store_to_asset_cache(context, raw_url, sanitized_url, "PUT", headers, file);
+
+        static constexpr StringLiteral guid_marker = "63ba93be-2084-4751-8b4b-d481c80d965c";
+
+        // base64 encoded IDs, generated with
+        // for I in @ A B C D E F G H I; do echo '"'$(echo "00${I}" | base64)'",' ; done
+        static const char* block_ids[10] = {
+            "MDBACg==",
```

Comment:
base64 Ids can be just number in length of 4. It don't need to be with any meaning.
for example 0000, 0001 ... 9999. More id's without base64 encode to generate them.

Reply:
No meaning assumed, but it order helps humans. The base64 encoding is required by  
https://learn.microsoft.com/en-us/rest/api/storageservices/put-block?tabs=microsoft-entra-id#uri-parameters  
and also sued in the examples, e.g.  
https://learn.microsoft.com/en-us/rest/api/storageservices/put-block-list?tabs=microsoft-entra-id#sample-request

Reply:
In the example you gave, the base64 ids also doesn't have a meaning or order ...
and my suggestion the base64 ids have an order, the decoded one also have some order in the first 5.

Reply:
The IDs identify blocks, and the block list defines the order of the blocks in the blob.

Which problem are you trying to talk about?

Reply:
> No meaning assumed, but order helps humans

Reply:
the string "0000" is already base64 encode. you can decode it and azure accept that ids'

Reply:
The string "0000" doesn't decode to a printable string. How do I know that there are no "forbidden characters"?
~~~
$ echo 0000 | base64 -d | strings
$
~~~

Reply:
because I test it on vcpkg ci and it accept it.

Reply:
The question is if it happens to work (empirically) or if it is part of the contract (design). If you cannot answer that with confidence, please leave it to reviewers that do know.

Reply:
From the link you share from the example:
'''
  <Latest>AAAAAA==</Latest>  
  <Latest>AQAAAA==</Latest>  
  <Latest>AZAAAA==</Latest>  
'''

These ids is also not a printable strings

Reply:
Well, finally making a point.

Reply:
Personally, I'm ok with the hardcoded IDs since we would always be encoding the same values (a counter of 1 to N). The only way I'd see us needing base64 encoding would be if we had arbitrary size blocks (user controlled), but that's still up to discussion with the rest of the team.

Reply:
It is easy to print numbers 0000-9999 which all decode to three bytes.

Reply:
@BillyONeal agrees with using hardcoded values and suggests the following struct:

```c++
struct BlockId {
    char result[9];
 
    constexpr BlockId(char ch) : result{'M', 'D', 'B', ch, 'C', 'g', '=', '=', '\0'} {
        if (ch < 'A' || ch > 'Z') { abort(); }
    }
};
```

Reply:
> It is easy to print numbers 0000-9999 which all decode to three bytes.

This is also fine.

Reply:
To be clear, I was just suggesting that as an option, not saying I think that change needs to be made :)